### PR TITLE
Ctcp 2402: Add status to message object

### PR DESCRIPTION
**Pull request by liandra-giacomine**

### CODE REVIEW

File: app/uk/gov/hmrc/transitmovements/controllers/MovementsController.scala [hunk 1]
```diff
@@ -160,7 +166,7 @@ class MovementsController @Inject() (
       .asPresentation
       .fold[Result](
         baseError => Status(baseError.code.statusCode)(Json.toJson(baseError)),
-        movements => if (movements.isEmpty) NotFound else Ok(Json.toJson(movements))
+        movements => { println(Json.toJson(movements)); if (movements.isEmpty) NotFound else Ok(Json.toJson(movements)) }
```
Comment (derekspalmer) [inline app/uk/gov/hmrc/transitmovements/controllers/MovementsController.scala:169]: println?

File: app/uk/gov/hmrc/transitmovements/models/MessageStatus.scala [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import play.api.libs.json.JsError
+import play.api.libs.json.JsString
+import play.api.libs.json.JsSuccess
+import play.api.libs.json.Json
+import play.api.libs.json.Reads
+import play.api.libs.json.Writes
+
+sealed trait MessageStatus
+
+object MessageStatus {
```
Comment (derekspalmer) [inline app/uk/gov/hmrc/transitmovements/models/MessageStatus.scala:28]: Are these as a result of the spike?

File: app/uk/gov/hmrc/transitmovements/models/MessageType.scala [hunk 3]
```diff
@@ -69,21 +69,21 @@ object MessageType {
   // *******************
 
   /** E_DEC_AMD (IE013) */
-  case object DeclarationAmendment extends DepartureRequestMessageType("IE013", "CC013C", "/xsd/CC013C.xsd")
+  final case object DeclarationAmendment extends DepartureRequestMessageType("IE013", "CC013C", "/xsd/CC013C.xsd")
```
Comment (derekspalmer) [inline app/uk/gov/hmrc/transitmovements/models/MessageType.scala:72]: objects are essentially final anyway, aren't they?

**Review by derekspalmer — COMMENTED** [review]
LGTM  :+1: - apart from println.

File: app/uk/gov/hmrc/transitmovements/controllers/MovementsController.scala (hunk 1 above)
Comment (liandra-giacomine) [reply app/uk/gov/hmrc/transitmovements/controllers/MovementsController.scala:169]: whoops - good spot, it was for debugging

File: app/uk/gov/hmrc/transitmovements/models/MessageStatus.scala (hunk 2 above)
Comment (liandra-giacomine) [reply app/uk/gov/hmrc/transitmovements/models/MessageStatus.scala:28]: Wasn't a spike but we had discussions in the chat/clarification

File: app/uk/gov/hmrc/transitmovements/models/MessageType.scala (hunk 3 above)
Comment (liandra-giacomine) [reply app/uk/gov/hmrc/transitmovements/models/MessageType.scala:72]: yeah - https://nrinaudo.github.io/scala-best-practices/adts/final_case_objects.html

**Review by derekspalmer — APPROVED** [review]

**Pull request merged**